server/elasticsearch/src/test/java/org/hawkular/apm/server/elasticsearch/SpanServiceElasticsearchTest.java

```diff
@@ -0,0 +1,247 @@
+        span.setBinaryAnnotations(Arrays.asList(binaryAnnotation));
+
+        storeAndWait(null, span);
+        Span spanFromDb = spanService.getSpan(null, span.getId());
+
+        assertSpansEquals(span, spanFromDb);
+    }
+
+    @Test
+    public void testGetChildren() throws StoreException, InterruptedException {
+        Span parentSpan = new Span();
+        parentSpan.setId("parent");
+        parentSpan.setName("foo parent");
+
+        storeAndWait(null, parentSpan);
+
+        Annotation annotation = new Annotation();
+        annotation.setValue("foo bar");
+        annotation.setTimestamp(123456789);
+        annotation.setEndpoint(createEndpoint("123.123.123.1233", 123, "foo service"));
+
+        BinaryAnnotation binaryAnnotation = new BinaryAnnotation();
+        binaryAnnotation.setEndpoint(createEndpoint("123.123.123.1233", 123, "foo service"));
+        binaryAnnotation.setValue("foo");
+        binaryAnnotation.setKey("foo key");
+        binaryAnnotation.setType(AnnotationType.I64);
+
+        Span span = new Span();
+        span.setId("id1");
+        span.setTraceId("traceId");
+        span.setParentId("parent");
+        span.setName("foo");
+        span.setDebug(true);
+        span.setTimestamp(1234456L);
+        span.setDuration(55468L);
+        span.setAnnotations(Arrays.asList(annotation));
+        span.setBinaryAnnotations(Arrays.asList(binaryAnnotation));
+
+        storeAndWait(null, span);
+
+        span = new Span();
+        span.setId("id2");
+        span.setParentId("parent");
+
+        storeAndWait(null, span);
+
+        List<Span> children = spanService.getChildren(null, "parent");
+
+        Assert.assertEquals(2, children.size());
+    }
+
+    @Test
+    public void testStoreWithIdSupplier() throws StoreException, InterruptedException {
+        final String clientIdsuffix = "-client";
+
+        Span clientSpan = new Span();
+        clientSpan.setId("foo");
+        clientSpan.setDuration(111);
+        clientSpan.setAnnotations(annotations());
+
+        storeAndWait(null, clientSpan, x -> x.getId() + clientIdsuffix);
+
+        Span span = new Span();
+        span.setId("foo");
+        span.setDuration(444);
+
+        storeAndWait(null, span, x -> x.getId());
+
+        Span spanFromDb = spanService.getSpan(null, "foo");
+        assertSpansEquals(span, spanFromDb);
+
+        spanFromDb = spanService.getSpan(null, "foo" + clientIdsuffix);
+        assertSpansEquals(clientSpan, spanFromDb);
+    }
+
+    @Test
+    public void testGetChildrenFromEmptyDB() {
+        spanService.getChildren(null, "id");
+    }
+
+    @Test(expected = IndexMissingException.class)
+    public void testGetSpanFromEmptyDB() {
+        spanService.getSpan(null, "id");
+    }
+
+    private void storeAndWait(String tenant, Span span) throws StoreException, InterruptedException {
+        spanService.storeSpan(tenant, Arrays.asList(span));
+        synchronized (this) {
+            wait(1000);
```

Comment:
By the way, which operations from the `storeSpan` is async? Just by looking at the code, I couldn't spot where it would be.